### PR TITLE
refactor: normalize definition of annotations and labels

### PR DESCRIPTION
**Pull request by supereagle**
**What this PR does / why we need it**:

Normalize definition of annotations and labels. Partial work of #848.

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @zhujian7 @cd1989 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/meta/annotations.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+	// AnnotationTenantInfo is the annotation key used for namespace to relate tenant information
+	AnnotationTenantInfo = "tenant.cyclone.io/info"
+
+	// AnnotationTenantStorageUsage is annotation to store storage usuage information
+	AnnotationTenantStorageUsage = "tenant.cyclone.io/storage-usage"
+
+	// AnnotationMetaNamespace is annotation applied to pod to specify the namespace where Workflow, WorkflowRun etc belong to.
+	// TODO(robin) What is better?
+	AnnotationMetaNamespace = "cyclone.io/meta-namespace"
```
Comment (supereagle) [inline pkg/meta/annotations.go]: Need to consider a better one.

**Comment by supereagle** [comment]
Need a better API group name.

/hold

File: pkg/meta/labels.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	LabelProjectName = "project.cyclone.dev/name"
+
+	// LabelWorkflowName is the label key used to indicate the workflow which the resources belongs to
+	LabelWorkflowName = "workflow.cyclone.dev/name"
+
+	// LabelWorkflowRunName is the label key used to indicate the workflowrun which the resources belongs to
+	LabelWorkflowRunName = "workflowrun.cyclone.dev/name"
+
+	// LabelWorkflowRunAccelerated is the label key used to indicate a workflowrun turned on acceleration
+	LabelWorkflowRunAccelerated = "workflowrun.cyclone.dev/accelerated"
```
Comment (supereagle) [inline pkg/meta/labels.go]: Rename `acceleration` to `accelerated`.  ping @cd1989  WDYT?

**Comment by supereagle** [comment]
Rename of API group name in manifest and cyclone web will be done in separated PRs.

File: pkg/meta/annotations.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+	// AnnotationTenantStorageUsage is annotation to store storage usuage information
+	AnnotationTenantStorageUsage = "tenant.cyclone.dev/storage-usage"
+
+	// AnnotationMetaNamespace is annotation applied to pod to specify the namespace where Workflow, WorkflowRun etc belong to.
+	// TODO(robin) What is better?
+	AnnotationMetaNamespace = "cyclone.dev/meta-namespace"
+
+	// AnnotationGCPod is annotation applied to pod to indicate whether the pod is used for GC purpose
+	AnnotationGCPod = "gc.cyclone.dev"
```
Comment (cd1989) [inline pkg/meta/annotations.go]: How about `pod.cyclone.dev/kind`, then `pod.cyclone.dev/kind=gc`

Comment (cd1989) [reply pkg/meta/annotations.go]: I think it's better to be label, instead of annotation.

File: pkg/meta/labels.go [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+	// LabelStageTemplate is the label key used to represent a stage is a stage template
+	LabelStageTemplate = "stage.cyclone.dev/template"
+
+	// LabelIntegrationType is the label key used to indicate type of integration
+	LabelIntegrationType = "integration.cyclone.dev/type"
+
+	// LabelIntegrationClusterSchedulable is the label key used to indicate the cluster is schedulable for workflowruns in this tenant
+	LabelIntegrationClusterSchedulable = "integration.cyclone.dev/cluster-schedulable"
```
Comment (cd1989) [inline pkg/meta/labels.go]: I'd prefer `schedulable-cluster`.

File: pkg/meta/annotations.go (hunk 3 above)
Comment (supereagle) [reply pkg/meta/annotations.go]: Discuss offline, will use `pod.kubernetes.io/created-by=cyclone` and `pod.kubernetes.io/kind=gc`

File: pkg/server/biz/accelerator/accelerate.go [hunk 5]
```diff
@@ -42,7 +43,7 @@ func NewAccelerator(tenant, project string, wfr *api.WorkflowRun) *Accelerator {
 	}
 }
 
-// Accelerate will check if the workflowrun has label 'workflowrun.cyclone.io/acceleration=true',
+// Accelerate will check if the workflowrun has label 'workflowrun.cyclone.dev/accelerated=true',
```
Comment (zhujian7) [inline pkg/server/biz/accelerator/accelerate.go]: `acceleration`

File: manifests/templates/cicd/cd.yaml [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
     name: cd-template
     labels:
-      cyclone.io/stage-template: "true"
-      cyclone.io/builtin: "true"
-      cyclone.io/scene: cicd
-      cyclone.io/template-kind: cd
+      stage.cyclone.dev/template: "true"
+      cyclone.dev/builtin: "true"
+      cyclone.dev/scene: cicd
+      cyclone.dev/template-kind: cd
```
Comment (zhujian7) [inline manifests/templates/cicd/cd.yaml]: should be `stage.cyclone.dev/template-kind`? please check others

**Pull request merged**